File: Leetcode 150/Array_String/12-integer-to-roman/integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        hash=[(1000, 'M'), (900, 'CM'), (500, 'D'), (400, 'CD'), (100, 'C'), (90, 'XC'), (50, 'L'), (40, 'XL'), (10, 'X'), (9, 'IX'), (5, 'V'), (4, 'IV'), (1, 'I')]
        string = []
        for a,b in hash:
            if num==0:
                break
            else:
                count = math.floor(num/a)
                string.append(b*count)
                num -= count*a
        return ''.join(string)

File: Leetcode 150/Array_String/12-integer-to-roman/integer-to-roman.java
class Solution {
    public String intToRoman(int num) {
        StringBuilder s = new StringBuilder();
        final int[] numbers = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        final String[] romans = {"M", "CM", "D",  "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        for(int i=0;i<numbers.length;i++)
        {
            if(num==0){
                break;
            }
            while(num>=numbers[i])
            {
                s.append(romans[i]);
                num -= numbers[i];
            }
        }
        return s.toString();
    }
}